st2reactor/st2reactor/cmd/timersengine.py

```diff
@@ -0,0 +1,88 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+import os
+import sys
+
+import eventlet
+from oslo_config import cfg
+
+from st2common import log as logging
+from st2common.constants.timer import TIMER_ENABLED_LOG_LINE, TIMER_DISABLED_LOG_LINE
+from st2common.logging.misc import get_logger_name_for_module
+from st2common.service_setup import setup as common_setup
+from st2common.service_setup import teardown as common_teardown
+from st2common.util.monkey_patch import monkey_patch
+from st2reactor.timer import config
+from st2reactor.timer.base import St2Timer
+
+monkey_patch()
+
+LOGGER_NAME = get_logger_name_for_module(sys.modules[__name__])
+LOG = logging.getLogger(LOGGER_NAME)
+
+
+def _setup():
+    common_setup(service='timer_engine', config=config, setup_db=True, register_mq_exchanges=True,
+                 register_signal_handlers=True)
+
+
+def _teardown():
+    common_teardown()
+
+
+def _kickoff_timer(timer):
+    timer.start()
+
+
+def _run_worker():
+    LOG.info('(PID=%s) TimerEngine started.', os.getpid())
+
+    timer = None
+
+    try:
+        timer_thread = None
+        if cfg.CONF.timer.enable or cfg.CONF.timersengine.enable:
```

Comment:
It would be a bit config is one is False and other is True, so perhaps we should be more explicit and throw in such scenario? Or?

Reply:
With default configs, you'll always have this behavior. So we have to actually detect if those variables are defined in /etc/st2/st2.conf. I think that's not really required. We have documented in upgrade notes. And when people upgrade, the configuration change diff will be shown to them too. So there are multiple checks and alerts to the user.

Reply:
One thing which is also config is that because we default one option to True, when user wants to disable the service they need to set **both** options to False.

I will document that (if not already).